base/cvd/cuttlefish/host/libs/metrics/metrics_orchestration.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+
+#include "cuttlefish/common/libs/utils/result.h"
+#include "cuttlefish/host/commands/cvd/instances/instance_group_record.h"
+
+namespace cuttlefish {
+
+std::string GenerateMetricsDirectoryFilepath(
+    const LocalInstanceGroup& instance_group);
+
+Result<void> SetUpMetrics(const std::string& metrics_directory);
+
+Result<void> GatherVmInstantiationMetrics(const std::string& metrics_directory);
```

Comment:
nit: Instead of having a function return a directory that the callers never use and simply pass to other functions of this header, consider an object oriented approach like this:
```
class MyBetterNamedMetricsClass {
 public:
  static Result<MyBetterNamedMetricsClass> Create(); 

  Result<void> GatherVmInstantiationMetrics();

 private:
  MyBetterNamedMetrics(std::string metrics_directory) : metrics_directory_(std::move(metrics_directory)) {}

  std::string metrics_directory_;
};
```